src/cacheUntilType.js

```diff
@@ -0,0 +1,51 @@
+import { OuterSubscriber } from 'rxjs/OuterSubscriber';
+import { subscribeToResult } from 'rxjs/util/subscribeToResult';
+
+
+export function cacheUntilType(type) {
+  return this.lift(new CacheUntilTypeOperator(type));
+}
+
+class CacheUntilTypeOperator {
```

Comment:
totally. FYI it does that for perf reasons cause _at the time_ closures were much slower than objects in v8. Since then they've released turbofan and shit...so it may no longer hold true. :man_shrugging: